velox/type/Type.h

```diff
@@ -628,7 +678,9 @@ class DecimalType : public ScalarType<KIND> {
       KIND == TypeKind::SHORT_DECIMAL ? 18 : 38;
 
   DecimalType(const uint8_t precision = 18, const uint8_t scale = 0)
-      : precision_(precision), scale_(scale) {
+      : precision_(precision),
+        scale_(scale),
+        parameters_{TypeParameter::of(precision), TypeParameter::of(scale)} {
```

Comment:
as a random thought: should this just be the struct's overloaded constructor? 

```
  parameters_{TypeParameter{precision}},
```

Reply:
That works. Let me know if I should change it like so.

Reply:
That would looks a bit more natural, IMHO, but it's really a nit. Up to you.